---
title: "Figure 8ab"
output: html_document
date: "2024-05-02"
---
```{r setup, include=FALSE, warning=FALSE}
library(ggpubr)
library(dplyr)
library(ggplot2)
library(Peptides)
library(motifStack)
library(pheatmap)
library(reshape)
library(ComplexHeatmap)
library(tidyverse)
library(viridis)
library(gridExtra)
library(grid)

```

```{r input, echo=FALSE, warning=FALSE}
data_dir<-"/data/"
master_file<-"master_maf.txt"
master_file_ndt<-"master_NDT.txt"
master<-read.delim(paste0(data_dir,master_file),sep="\t",header=TRUE)
master_ndt<-read.delim(paste0(data_dir,master_file_ndt),sep="\t",header=TRUE)

```

```{r input2,echo=FALSE, warning=FALSE}
clinical_data<-read.delim(paste0(data_dir,"bms153dat-021918.csv"),sep=",",header=TRUE)
clinical_data_response<-unique(clinical_data[,c("PatientID","myBOR","OS_event","OS_time_Mon","PFS_event","PFS_time")])
clinical_data_response$myBOR[which(is.na(clinical_data_response$myBOR))]<-"PD"
clinical_data_response$color<-NA
clinical_data_response$color[which(clinical_data_response$myBOR == "CR/PR")]<-"green"
clinical_data_response$color[which(clinical_data_response$myBOR == "SD")]<-"orange"
clinical_data_response$color[which(clinical_data_response$myBOR == "PD")]<-"red"
clinical_data_response$myBOR<-gsub("CR/PR","PR",clinical_data_response$myBOR)
```

```{r 8a, echo=FALSE}
master<-master[which(master$Pep_Length==9),]
master<-master[which(master$MT.Allele=="A0201"),]
master$tetramer[which(master$Binder=="NON_BINDER")]<-"NON_BINDER"
df<-as.data.frame(as.matrix(table(master$Patient,master$tetramer)))

ggplot(df,aes(x=Var1,y=Freq,fill=Var2)) + geom_bar(stat = "identity",width = 0.5) +theme_classic()+scale_fill_manual(values=c("pink", "green2", "blue2"))


```
```{r echo=FALSE}
data("AAdata") ## load amino acids from Peptides package

aa = names(AAdata$Hydrophobicity$Kidera) ## 20 amino acids

## one-hot encode peptide sequence
## for a given peptide, this will return a data frame of dimensions 1 X 180 (assuming the peptide is a 9mer)
## each column of the encoding is a particular amino acid at a particular position (1 if present, 0 if not)
peptide_encode = function(peptide){
  x = unlist(strsplit(peptide, ""))
  
  d = as.data.frame("dummy")
  colnames(d)[1] = "dummy"
  
  for(i in seq(length(x))){
    # print(i)
    x2 = as.data.frame(t(as.integer(aa %in% x[i])))
    colnames(x2) = aa
    colnames(x2) = paste(paste(paste("P", i, sep = ""), colnames(x2), sep = "_"))
    d = cbind(d,x2)
  }
  d = d[,2:ncol(d)]
  return(d)
}
```


```{r echo=FALSE}
master<-read.delim("master_maf.txt",header = TRUE, sep = "\t")
master<-master[which(master$Pep_Length==9),]
master<-master[which(master$MT.Allele=="A0201"),]
#master<-master[which(master$Variant_Type=="SNP"),]
peptides_sequence<-master$MT.Peptide.x

### creating logo plots 
pos_peptides<-master$MT.Peptide.x[which(master$tetramer=="TETRAMER+")]
neg_peptides<-master$MT.Peptide.x[which(master$tetramer=="TETRAMER-")]

bigmat = data.frame()
for(i in peptides_sequence){
  pe = peptide_encode(i)
  bigmat = rbind(bigmat, pe)
}

bigmat$tet_pos_or_neg = master$tetramer
bigmat<-na.omit(bigmat)
bigmat$tet_pos_or_neg = ifelse(bigmat$tet_pos_or_neg=="TETRAMER+",1,0)

bigmat$tet_pos_or_neg<-as.factor(bigmat$tet_pos_or_neg)


bigmat_tet_pos<-bigmat[which(bigmat$tet_pos_or_neg==1),]
bigmat_tet_neg<-bigmat[which(bigmat$tet_pos_or_neg==0),]
bigmat_pos2<-melt(bigmat_tet_pos[1:180])
bigmat_neg2<-melt(bigmat_tet_neg[1:180])

bigmat_sum_pos<-data.frame(matrix(nrow=180, ncol=2))
colnames(bigmat_sum_pos)<-c("position","value")
bigmat_sum_pos$position<-unique(bigmat_pos2$variable)
for(i in unique(bigmat_pos2$variable)){
  pos<-sum(bigmat_pos2$value[which(bigmat_pos2$variable==i)])
  bigmat_sum_pos$value[which(bigmat_sum_pos$position==i)]<-pos
}

bigmat_sum_neg<-data.frame(matrix(nrow=180, ncol=2))
colnames(bigmat_sum_neg)<-c("position","value")
bigmat_sum_neg$position<-unique(bigmat_neg2$variable)
for(i in unique(bigmat_neg2$variable)){
  pos<-sum(bigmat_neg2$value[which(bigmat_neg2$variable==i)])
  bigmat_sum_neg$value[which(bigmat_sum_neg$position==i)]<-pos
}
```

```{r seqLogo,echo=FALSE}
bigmat_sum <- data.frame(position=bigmat_sum_pos$position,value=as.numeric(((bigmat_sum_pos$value/44))/((bigmat_sum_neg$value/108))))
bigmat_sum$value<-gsub("NaN",0,bigmat_sum$value)
bigmat_sum$value<-gsub("Inf",4,bigmat_sum$value)
bigmat_sum$amino_acid<-str_split_fixed(bigmat_sum$position,"_",2)[,2]
bigmat_sum$pos<-str_split_fixed(bigmat_sum$position,"_",2)[,1]
bigmat_sum<-bigmat_sum[,2:4]

heatmap_df<-spread(bigmat_sum,key=pos,value=value)

rownames(heatmap_df)<-heatmap_df$amino_acid
heatmap_df<-heatmap_df[,2:10]
heatmap_df[,1:9]<-lapply(heatmap_df[,1:9],as.numeric)
cols<-c("#FDE725FF","#DCE319FF","white","#8856A7","#810F7C")
#cols<-colorRampPalette((c("#FDE725FF", "white", "#810F7C")))(5)
pheatmap::pheatmap(heatmap_df, cluster_cols = FALSE, scale = "column",color =cols)

motif_df<-data.matrix(heatmap_df)
motif_df<-pcm2pfm(motif_df)
motif<-new("pfm", mat=motif_df, name="CAP",
           color=colorset(alphabet="AA",colorScheme="chemistry"))
plot(motif,ic.scale=FALSE,)
```

